MIGRATION-V9.md

```diff
@@ -0,0 +1,324 @@
+
+### Discovery v2 Models
+
+#### Models Added
+
+`AnalyzedDocument`, `AnalyzeDocumentOptions`, `AnalyzedResult`
+
+#### Models Removed
+
+`QueryNoticesResult`
+
+#### Model Properties Changed
+
+`CollectionDetails`
+* Changed `collectionId` and `created` properties to read only
+
+`TrainingExample`
+* Changed `created` and `updated` properties to read only
+
+`TrainingQuery`
+* Changed `created`, `queryId` and `updated` properties to read only
+
+### Language Translator v3 Instance
+
+* No changes
+
+### Language Translator v3 Models
+
+#### Model Properties Changed
+
+`ListDocumentsOptions`, `ListIdentifiableLanguagesOptions`, `ListLanguagesOptions`
+* Removed Builder
+
+### Natural Language Classifier v1 Instance
+
+* No changes
+
+### Natural Language Classifier v1 Models
+
+#### Model Properties Changed
+
+`ListClassifiersOptions`
+* Removed Builder
+
+### Natural Language Understanding v1 Instance
+
+* No changes
+
+### Natural Language Understanding v1 Models
+
+#### Models Added
+
+`FeaturesResultsMetadata`
+
+#### Models Removed
+
+`MetadataOptions`, `AnalysisResultsMetadata`
+
+#### Model Properties Changed
+
+`AnalysisResults`
+* Changed `metadata` property type from `AnalysisResultsMetadata` to `FeaturesResultsMetadata`
+
+`Features`
+* Changed `metadata` property type from `MetadataOptions` to `Map<String, Object>`
+
+`ListModelsOptions`
+* Remove Builder
+
+### Personality Insights v3 Instance
+
+* No changes
+
+### Personality Insights v3  Models
+
+* No changes
+
+### Speech To Text v1 Instance
+
+* `recognizeUsingWebSocket()`: Changed `recognizeOptions` parameter type from `RecognizeOptions` to `RecognizeWithWebsocketsOptions`
+
+
+### Speech To Text v1 Models
+
+#### Models Added
+
+`RecognizeWithWebsocketsOptions`
+
+#### Model Properties Changed
+
+`CheckJobsOptions`, `ListModelOptions`
+* Removed Builder
+
+`RecognizeOptions`
+* Removed `interimResults`, `processingMetrics`, `processingMetricsInterval`
+properties (Moved to `RecognizeWithWebsocketsOptions`)
+
+### Text to Speech v1 Instance
+
+* No changes
+
+### Text to Speech v1 Models
+
+#### Models Added
+
+`CreateCustomModelOptions`, `CustomModel`, `CustomModels`, `DeleteCustomModelOptions`, `GetCustomModelOptions`, `ListCustomModelsOptions`, `UpdateCustomModelOptions`
+
+#### Models Removed
+
+`CreateVoiceModelOptions`, `DeleteVoiceModelOptions`, `GetVoiceModelOptions`, `ListVoiceModelsOptions`, `UpdateVoiceModelOptions`, `VoiceModel`, `VoiceModels`
+
+#### Model Properties Changed
+
+`ListVoicesOptions`
+* Remove Builder
+
+`Voice`
+* Change return type of `getCustomization()` from `VoiceModel` to `CustomModel`
+
+### Tone Analyzer v3 Instance
+
+* No changes
+
+### Tone Analyzer v3 Models
+
+* No changes
+
+### Visual Recognition v3 Instance
+
+* No changes
+
+### Visual Recognition v3 Models
+
+* No changes
+
+### Visual Recognition v4 Instance
```

Comment:
These should be called `Service` rather than `Instance`

> Visual Recognition v4 Service

Reply:
Fixed!